aea/skills/base.py

```diff
@@ -912,3 +765,348 @@ def _print_warning_message_for_non_declared_skill_components(
+                    component_class_or_none: Optional[Type[SkillComponent]] = next(
+                        (
+                            actual_class
+                            for actual_class_name, actual_class in classes_in_path
+                            if actual_class_name == class_name
+                        ),
+                        None,
+                    )
+                    enforce(
+                        component_class_or_none is not None,
+                        self._get_error_message_prefix()
+                        + f"Cannot find class '{class_name}' for component '{component_id}' of type '{self._type_to_str(component_type)}' of skill '{self.configuration.public_id}' in module {path}",
+                    )
+                    component_class = cast(
+                        Type[SkillComponent], component_class_or_none
+                    )
+                    used_classes.add(component_class)
+                    result.append(
+                        _SkillComponentLoadingItem(
+                            component_id,
+                            component_config,
+                            component_class,
+                            component_type,
+                        )
+                    )
+                else:
+                    # process the configuration at the end of the loop
+                    not_resolved_configurations[
+                        (component_type, component_id)
+                    ] = component_config
+
+        for (component_type, component_id), component_config in copy(
+            not_resolved_configurations
+        ).items():
+            class_name = component_config.class_name
+            classes_by_type = class_index.get(class_name, {})
+            enforce(
+                class_name in class_index and component_type in classes_by_type,
+                self._get_error_message_prefix()
+                + f"Cannot find class '{class_name}' for skill component '{component_id}' of type '{self._type_to_str(component_type)}'",
+            )
+            classes = classes_by_type[component_type]
+            not_used_classes = classes.difference(used_classes)
+            enforce(
+                not_used_classes != 0,
+                f"Cannot find class of skill '{self.configuration.public_id}' for component configuration '{component_id}' of type '{self._type_to_str(component_type)}'.",
+            )
+            enforce(
+                len(not_used_classes) == 1,
+                self._get_error_message_ambiguous_classes(
+                    class_name, not_used_classes, component_type, component_id
+                ),
+            )
+            not_used_class = list(not_used_classes)[0]
+            result.append(
+                _SkillComponentLoadingItem(
+                    component_id, component_config, not_used_class, component_type
+                )
+            )
+            used_classes.add(not_used_class)
```

Comment:
the non-used class now becomes used. This information can be used later to print a warning message for skill components not used (not done yet).